pkg/rtc/localparticipant.go

```diff
@@ -0,0 +1,347 @@
+package rtc
```

Comment:
`LocalParticipant` split out to handle all the pending tracks bits. This is where there will be a bunch of conflicts with session migration PR which I will resolve.

Ideally, we will make an interface of this and make a NoOp impl and use it in cloud